src/target/riscv/riscv.c

```diff
@@ -451,7 +452,7 @@ static int riscv_init_target(struct command_context *cmd_ctx,
 
 	if (bscan_tunnel_ir_width != 0) {
 		assert(target->tap->ir_length >= 6);
-		uint32_t ir_user4_raw = 0x23 << (target->tap->ir_length - 6);
+		uint32_t ir_user4_raw = bscan_tunnel_ir_id != 0 ? bscan_tunnel_ir_id : 0x23 << (target->tap->ir_length - 6);
```

Comment:
```suggestion
		uint32_t ir_user4_raw = bscan_tunnel_ir_id << (target->tap->ir_length - 6);
```

Reply:
The original code was "0x23 << (target->tap->ir_length - 6);" I may be wrong but I think the intend of that shift was to cover some other device from xilinx  which would follow the rule  0x23 << (ir_length - 6), which is would be very specific.

If it is the case, i think that shift should not be applied to user specified bscan_tunnel_ir_id.

So, 
```c
uint32_t ir_user4_raw = bscan_tunnel_ir_id << (target->tap->ir_length - 6);
```
would be problematic.

Reply:
Thanks. I missed that detail.

I had hoped there was a way to default to the current behavior without the extra logic, but if the current behavior depends on ir_length then that's really not doable and your implementation is our best option.